src/plans/stepsize.jl

```diff
@@ -959,3 +959,53 @@ function get_last_stepsize(
 )
     return step.last_stepsize
 end
+
+# stepsize storage
+
+stepsize_storage_points(::Stepsize) = Tuple{}
+
+stepsize_storage_vectors(::Stepsize) = Tuple{}
+
+"""
+    StepsizeStorage{TS<:Stepsize,TStorage<:StoreStateAction}
+
+Storage for stepsize kept between iterations of a solver.
+"""
+mutable struct StepsizeStorage{TS<:Stepsize,TStorage<:StoreStateAction,T<:Number} <:
```

Comment:
While I like this – what is stored here?

alpha and last_f_p (is that the last cost?) could be stored in the wrapper – they are stored there for the usual step sizes anyways.

Reply:
What I do not like of this struct for now is – introducing it here says: This is a Manopt-thing. But it is just used in the wrapper and not in the Stepsizes of Manopt. So is it a Manopt-thing?
Why is it then only required in QN that one uses it? I would have expected the old ones to work without this?

Reply:
This is intended to store things stepsize-related information between iteration of an optimization algorithm. Right now it is only used for LineSearches.jl but I expect that it would also make sense to use it in "native" Manopt stepsize algorithms. It is not QN-specific but it's the one solver where I cared about using LineSearches.jl and `StepsizeStorage` is needed for those.

Alpha stores stepsize from the previous line search and last_f_p stores value of objective at the point from which the last line search started (that's what Optim.jl does). They could be stored in the wrapper, I put it here because its something stored and we need `StepsizeStorage` for p_tmp, X_tmp and Y_tmp anyway.

Reply:
then the fields should be called `last_stepsize` and `initial_cost`.

I think the idea is a good one since we have the storage idea anyways and sure others could use that as well later.

Reply:
I've updated this.